### PR TITLE
Response renaming

### LINKED ISSUE #3271: Rename RestResponse class and RestResponse.body method
After discussion in the architecture board, it was decided to use `Response` for the base `RestResponse` interface, and to find a better name for `RestResponse.body()`. The current name used by .net is 'Result', but @KrzysztofCwalina is not convinced that is the best name either. We should work together to have a good and intuitive name.

### PR DESCRIPTION
**Pull request by JonathanGiles**
This commit makes three primary changes:
1. Renames all rest response classes to remove the 'Rest' in the name. The argument for this is that these are the most common return types, so they should be shorter. Other return types, when introduced for supporting other services, can have longer, more specific names. This resolves #3271.
2. Renames 'Response.body()' to 'Response.result()'. This resolves #3271.
3. Introduces a new 'AzureException' super class, and renames 'RestException' to 'ServiceRequestException' in the new com.azure.common.exception package. (Note: @sima-zhu This means I've started the work outlined in #3229, but you should finish it off).

### CODE REVIEW

**Comment by JonathanGiles** [comment]
@anuchandy A unit test is failing on mac OS after these changes were applied. However, from looking at the code, I suspect it is more a case of time sensitive code not quite being the same on all systems. For example, the test passes on my MacBook Pro, but fails in Azure Pipelines on MacOS. Do you have any thoughts? Here is the JUnit failure:

```
[ERROR] createAsyncWithAzureAsyncOperationAndPollsWithDelay(com.azure.common.mgmt.AzureProxyTests)  Time elapsed: 0.276 s  <<< FAILURE!
java.lang.AssertionError: expected:<1> but was:<2>
	at com.azure.common.mgmt.AzureProxyTests.createAsyncWithAzureAsyncOperationAndPollsWithDelay(AzureProxyTests.java:406)

[ERROR] Failures: 
[ERROR]   AzureProxyTests.createAsyncWithAzureAsyncOperationAndPollsWithDelay:406 expected:<1> but was:<2>
```

**Comment by JonathanGiles** [comment]
@anuchandy I have added an `@Ignore` annotation to the specific test that is failing, because it seems to be a timing-related issue rather than a bug in the code. I will file a follow-up issue to resolve this issue.

File: common/azure-common/src/main/java/com/azure/common/exception/AzureException.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * The base exception type for all Azure-related exceptions.
+ *
+ * @see ServiceRequestException
+ */
+public class AzureException extends RuntimeException {
+
+    /**
+     * Initializes a new instance of the AzureException class.
+     *
+     * @param message the exception message or the response content if a message is not available
```
Comment (anuchandy) [inline common/azure-common/src/main/java/com/azure/common/exception/AzureException.java]: Should we delete "or the response content if a message is not available" from description? -> `AzureException` is root exception type, so not always based on service `response`.

**Comment by anuchandy** [comment]
@JonathanGiles I will look into glitching test, ok to ignore it for now.

File: common/azure-common/src/main/java/com/azure/common/exception/AzureException.java (hunk 1 above)
Comment (JonathanGiles) [reply common/azure-common/src/main/java/com/azure/common/exception/AzureException.java]: Good point - I'll fix that. Thanks.

**Comment by JonathanGiles** [comment]
The current discussion in .NET is to call the `Response.body` / `Response.result` field `Response.value` instead. Is there any feedback anyone wants to offer on this, or should I go ahead and rename from `result` to `value` for consistency with .NET?

**Comment by JonathanGiles** [comment]
I have gone ahead with the `body` -> `result` -> `value` renaming on the `Response` interface.

**Pull request merged**